packages/mobile/src/send/SendConfirmation.tsx

```diff
@@ -89,6 +95,44 @@ function SendConfirmation(props: Props) {
 
   const dispatch = useDispatch()
 
+  const secureSendPhoneNumberMapping = useSelector(secureSendPhoneNumberMappingSelector)
+  const e164NumberToAddress = useSelector(e164NumberToAddressSelector)
+
+  const addressValidationType = getAddressValidationType(
+    paramRecipient,
+    secureSendPhoneNumberMapping
+  )
+  const validatedRecipientAddress = getSecureSendAddress(
+    paramRecipient,
+    secureSendPhoneNumberMapping
+  )
+  const recipient: Recipient = useMemo(() => {
+    if (!paramRecipient.address && paramRecipient.e164PhoneNumber) {
+      const recipientAddress = getAddressFromPhoneNumber(
+        paramRecipient.e164PhoneNumber,
+        e164NumberToAddress,
+        secureSendPhoneNumberMapping,
+        undefined
+      )
+      return {
+        ...paramRecipient,
+        // Setting it explicitly so Typescript doesn't complain
+        e164PhoneNumber: paramRecipient.e164PhoneNumber,
+        address: recipientAddress ?? undefined,
+      }
+    }
+    return paramRecipient
+  }, [paramRecipient])
```

Comment:
Feels a bit weird to do this here, should we handle this somewhere else?

Reply:
I moved it to a hook, but I'm not sure if that's what you had in mind 🤔